### PR TITLE
fix: add balance left for tx approval: staking

### PR DESCRIPTION
**Pull request by Sharqiewicz**
### What:

When user decides to stakes with "MAX" button, the entire PEN balance is considered in that transaction leaving no tokens for the user to execute the transaction i.e no gas fees to execute the transaction

### How:

Add BALANCE_LEFT_FOR_TX_APPROVAL constant variable.

Closes: #296

### CODE REVIEW

File: src/components/Form/Amount/index.tsx [hunk 1]
```diff
@@ -45,7 +46,7 @@ const Amount = ({ className, register, max, setValue, assetSuffix, error }: Amou
             </button>
             <button
               className="text-accent-content underline hover:opacity-70 mx-1 font-semibold"
-              onClick={() => setValue(roundNumber(Number(max)))}
+              onClick={() => setValue(roundNumber(Number(max) - BALANCE_LEFT_FOR_TX_APPROVAL))}
```
Comment (ebma) [inline src/components/Form/Amount/index.tsx:49]: A user that has `0.00` tokens available will now enter `-0.5` when clicking on the 'MAX' button. Let's maybe wrap this with
```suggestion
              onClick={() => setValue(Math.max(roundNumber(Number(max) - BALANCE_LEFT_FOR_TX_APPROVAL, 0))}
```

**Review by ebma — CHANGES_REQUESTED** [review]

File: src/components/Form/Amount/index.tsx (hunk 1 above)
Comment (Sharqiewicz) [reply src/components/Form/Amount/index.tsx]: @ebma updated ✅

**Review by ebma — APPROVED** [review]
LGTM 👍

**Pull request merged**